tests/register/register.go

```diff
@@ -52,20 +51,20 @@ func Register(tType TestType, t types.Test) {
 		{types20.MaxVersion, types21.MaxVersion, types22.MaxVersion, types23.MaxVersion},
 	}
 
-	version, semverErr := semver.NewVersion(t.ConfigMinVersion)
+	test := types.DeepCopy(t)
+	version, semverErr := semver.NewVersion(test.ConfigMinVersion)
+	test.ReplaceAllVersionVars(test.ConfigMinVersion)
+	register(tType, test) // some tests purposefully don't have config version
 
-	register(tType, t)
 	if semverErr == nil && version != nil && t.ConfigMinVersion != "" {
-		initalVersion := *version
 		for *version != configVersions[version.Major][len(configVersions[version.Major])-1] {
-			t := types.DeepCopy(t)
+			test = types.DeepCopy(t)
 
 			version.BumpMinor()
 			version = &configVersions[version.Major][version.Minor]
 
-			t.Name += " " + version.String()
-			t.Config = strings.Replace(t.Config, initalVersion.String(), version.String(), -1)
-			register(tType, t)
+			test.ReplaceAllVersionVars(version.String())
```

Comment:
This commit should be merged into the previous.

Reply:
I left it this way to have a distinction between the first commit which adds the backward compatibility vs the second commit which adds the $version stuff. Does it make sense to still merge them?

Reply:
To add to ^, the first commit updates the config with different versions. The second commit adds the ReplaceAllVersionVars() which updates $version with the different versions.

Reply:
I would merge them.

Reply:
@ajeddeloh merge both commits together?

Reply:
Done ^